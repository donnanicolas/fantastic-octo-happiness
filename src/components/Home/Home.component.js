// @flow
import React, { Component } from 'react';

import { FlexContainer, FlexItem } from '../Flex';
import SwipeableCards from '../SwipeableCards';

type Props = {
  user?: User,
  tracks: Track[],
  like: (track: Track) => any,
  userLoading: boolean,
  tracksLoading: boolean
};

class Home extends Component {

  props: Props;

  handleChange = ({ target: { value } }: SyntheticInputEvent) =>
    this.setState(state => ({ search: value }));

  handleLike = (track: Track) =>
    this.props.like(track);

  render() {
    const { user = {}, tracks, userLoading, tracksLoading } = this.props;
    const loading = userLoading || tracksLoading;

    return (
      <FlexContainer justifyContent="center">
        <FlexItem flex="0 1 240">
          {loading
            ? <h2>Loading ...</h2>
            : (
              <div>
                Hello {user.name}
                {tracks.length !== 0 &&
                  <SwipeableCards tracks={tracks} onSwipedRight={this.handleLike} />
                }
              </div>
            )
          }
        </FlexItem>
      </FlexContainer>
    )
  }
}

export default Home;