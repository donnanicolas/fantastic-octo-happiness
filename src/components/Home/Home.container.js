// @flow
import { connect } from 'react-redux';
import type { Dispatch } from 'redux';

import { likeTrack } from '../../actions/track';
import { getUser, getLoading as getUserLoading } from '../../reducers/user';
import { getTracks, getLoading as getTracksLoading } from '../../reducers/track';

import Home from './Home.component';

const mapStateToProps = (state: State) => ({
  user: getUser(state),
  tracks: getTracks(state),
  userLoading: getUserLoading(state),
  tracksLoading: getTracksLoading(state),
});

const mapDispatchToState = (dispatch: Dispatch<GenericAction>) => ({
  like: (track: Track) => dispatch(likeTrack(track)),
});

const container = connect(mapStateToProps, mapDispatchToState);
export default container(Home);
