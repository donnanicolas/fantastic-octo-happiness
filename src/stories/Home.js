// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Home from '../components/Home/Home.component';

import tracks from '../fixtures/tracks';

const Wrapper = ({ children }) =>
  <div style={{ height: '100%', backgroundColor: '#000' }}>
    {children}
  </div>

const addStories = () => {
  storiesOf('Home', module)
    .add('loading', () =>
      <Wrapper>
        <Home
          userLoading={true}
          tracksLoading={true}
          recommendTracks={action('recommendTracks')}
          tracks={tracks}
          like={action('like')}
        />
      </Wrapper>)
    .add('with tracks', () =>
      <Wrapper>
        <Home
          userLoading={false}
          tracksLoading={false}
          recommendTracks={action('recommendTracks')}
          tracks={tracks}
          like={action('like')}
        />
      </Wrapper>
    )
    .add('without tracks', () =>
      <Wrapper>
        <Home
          userLoading={false}
          tracksLoading={false}
          recommendTracks={action('recommendTracks')}
          tracks={[]}
          like={action('like')}
        />
      </Wrapper>
    )
};

export default addStories;