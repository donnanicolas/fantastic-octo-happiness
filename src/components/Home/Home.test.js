import React from 'react';
import Home from './Home.component';
import renderer from 'react-test-renderer';

const tracks = [
  {
    id: '1OJxI8lIWRqBvouJxW1nzN',
    name: 'Subwoofer Lullaby',
    popularity: 50,
    album: 'An Album',
    previewUrl: 'https://p.scdn.co/mp3-preview/872f1c73287e399cad17e225d8a8bf80581106af?cid=89b92773702c4e36a8014a880e7a9b7d',
    image: 'https://i.scdn.co/image/81aa003fb60a031abe1f72253cf663cff201ad48',
    artists: [{
      id: '4uFZsG1vXrPcvnZ4iSQyrx',
      name: 'C418',
    }],
  },
  {
    id: '0g8C2klxNYbNZS5VlobCxL',
    name: 'Any Which Way - Martin Buttrich Red Mix',
    popularity: 4,
    album: 'Any Which Way EP',
    previewUrl: 'https://p.scdn.co/mp3-preview/21d8989600ba902d0f0800bda40b4d8d3be717b0?cid=89b92773702c4e36a8014a880e7a9b7d',
    image: 'https://i.scdn.co/image/1e732874b19b8853dccd29a0e68959af87c8c1d5',
    artists: [{
      id: '0GJpYdmVCgg90TkyB1nB1y',
      name: 'Stacey Pullen',
    }],
  }
];

it('renders the Home page without tracks', () => {
  const tree = renderer.create(
    <Home tracks={[]}/>
  ).toJSON()
  expect(tree).toMatchSnapshot();
});

it('renders the Home page with tracks', () => {
  const tree = renderer.create(
    <Home tracks={tracks}/>
  ).toJSON()
  expect(tree).toMatchSnapshot();
});

