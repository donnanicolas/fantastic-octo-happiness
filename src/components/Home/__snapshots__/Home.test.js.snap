// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home page 1`] = `
<div
  className="gqseAW"
>
  <div
    className="ejnNlj"
  >
    <div>
      Hello 
    </div>
  </div>
</div>
`;
